### PR TITLE
HDDS-2180. Add Object ID and update ID on VolumeList Object.

### PR DESCRIPTION
**Pull request by anuengineer**
https://issues.apache.org/jira/browse/HDDS-2180
Adds Object ID and Update ID to VolumeList object.

### CODE REVIEW

File: hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/volume/OMVolumeRequest.java [hunk 1]
```diff
@@ -95,13 +100,18 @@ protected VolumeList addVolumeToOwnerList(VolumeList volumeList,
     }
 
     List<String> prevVolList = new ArrayList<>();
+    long objectID = 100;
```
Comment (xiaoyuyao) [inline hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/volume/OMVolumeRequest.java]: Is there a reason for initialize objectID to 100 here for new volume?

File: hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/volume/OMVolumeRequest.java [hunk 2]
```diff
@@ -95,13 +100,18 @@ protected VolumeList addVolumeToOwnerList(VolumeList volumeList,
...
     if (volumeList != null) {
       prevVolList.addAll(volumeList.getVolumeNamesList());
+      objectID = volumeList.getObjectID();
```
Comment (xiaoyuyao) [inline hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/volume/OMVolumeRequest.java]: Here we always use the objectID of the new volumeList to replace the existing one on line 113, is this expected?

File: hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/volume/OMVolumeRequest.java (hunk 1 above)
Comment (anuengineer) [reply hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/volume/OMVolumeRequest.java]: No, thanks I will fix that, I discovered it from a test code path, and left over from the fix. I will get this fixed.

File: hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/volume/OMVolumeRequest.java (hunk 2 above)
Comment (anuengineer) [reply hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/volume/OMVolumeRequest.java]: Yes, for some reason, when we replace a VolumeList, we do it like this. We read the current list, remove or add a new volume, create a new object and then write it back. This is reality nothing but a small change in the list of volumes. So I keep the same object ID as if Object is same.

**Comment by hadoop-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 43 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | dupname | 1 | No case conflicting files found. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| +1 | test4tests | 0 | The patch appears to include 5 new or modified test files. |
||| _ trunk Compile Tests _ |
| 0 | mvndep | 14 | Maven dependency ordering for branch |
| -1 | mvninstall | 30 | hadoop-hdds in trunk failed. |
| -1 | mvninstall | 26 | hadoop-ozone in trunk failed. |
| -1 | compile | 21 | hadoop-hdds in trunk failed. |
| -1 | compile | 15 | hadoop-ozone in trunk failed. |
| +1 | checkstyle | 49 | trunk passed |
| +1 | mvnsite | 0 | trunk passed |
| +1 | shadedclient | 834 | branch has no errors when building and testing our client artifacts. |
| -1 | javadoc | 21 | hadoop-hdds in trunk failed. |
| -1 | javadoc | 19 | hadoop-ozone in trunk failed. |
| 0 | spotbugs | 930 | Used deprecated FindBugs config; considering switching to SpotBugs. |
| -1 | findbugs | 31 | hadoop-hdds in trunk failed. |
| -1 | findbugs | 19 | hadoop-ozone in trunk failed. |
||| _ Patch Compile Tests _ |
| 0 | mvndep | 18 | Maven dependency ordering for patch |
| -1 | mvninstall | 35 | hadoop-hdds in the patch failed. |
| -1 | mvninstall | 29 | hadoop-ozone in the patch failed. |
| -1 | compile | 24 | hadoop-hdds in the patch failed. |
| -1 | compile | 19 | hadoop-ozone in the patch failed. |
| -1 | cc | 24 | hadoop-hdds in the patch failed. |
| -1 | cc | 19 | hadoop-ozone in the patch failed. |
| -1 | javac | 24 | hadoop-hdds in the patch failed. |
| -1 | javac | 19 | hadoop-ozone in the patch failed. |
| +1 | checkstyle | 56 | the patch passed |
| +1 | mvnsite | 0 | the patch passed |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedclient | 738 | patch has no errors when building and testing our client artifacts. |
| -1 | javadoc | 21 | hadoop-hdds in the patch failed. |
| -1 | javadoc | 20 | hadoop-ozone in the patch failed. |
| -1 | findbugs | 31 | hadoop-hdds in the patch failed. |
| -1 | findbugs | 20 | hadoop-ozone in the patch failed. |
||| _ Other Tests _ |
| -1 | unit | 28 | hadoop-hdds in the patch failed. |
| -1 | unit | 24 | hadoop-ozone in the patch failed. |
| +1 | asflicense | 33 | The patch does not generate ASF License warnings. |
| | | 2364 | |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.1 Server=19.03.1 base: https://builds.apache.org/job/hadoop-multibranch/job/PR-1526/1/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/1526 |
| Optional Tests | dupname asflicense compile cc mvnsite javac unit javadoc mvninstall shadedclient findbugs checkstyle |
| uname | Linux 19a38fab1e05 4.15.0-58-generic #64-Ubuntu SMP Tue Aug 6 11:12:41 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / bdaaa3b |
| Default Java | 1.8.0_222 |
| mvninstall | https://builds.apache.org/job/hadoop-multibranch/job/PR-1526/1/artifact/out/branch-mvninstall-hadoop-hdds.txt |
| mvninstall | https://builds.apache.org/job/hadoop-multibranch/job/PR-1526/1/artifact/out/branch-mvninstall-hadoop-ozone.txt |
| compile | https://builds.apache.org/job/hadoop-multibranch/job/PR-1526/1/artifact/out/branch-compile-hadoop-hdds.txt |
| compile | https://builds.apache.org/job/hadoop-multibranch/job/PR-1526/1/artifact/out/branch-compile-hadoop-ozone.txt |
| javadoc | https://builds.apache.org/job/hadoop-multibranch/job/PR-1526/1/artifact/out/branch-javadoc-hadoop-hdds.txt |
| javadoc | https://builds.apache.org/job/hadoop-multibranch/job/PR-1526/1/artifact/out/branch-javadoc-hadoop-ozone.txt |
| findbugs | https://builds.apache.org/job/hadoop-multibranch/job/PR-1526/1/artifact/out/branch-findbugs-hadoop-hdds.txt |
| findbugs | https://builds.apache.org/job/hadoop-multibranch/job/PR-1526/1/artifact/out/branch-findbugs-hadoop-ozone.txt |
| mvninstall | https://builds.apache.org/job/hadoop-multibranch/job/PR-1526/1/artifact/out/patch-mvninstall-hadoop-hdds.txt |
| mvninstall | https://builds.apache.org/job/hadoop-multibranch/job/PR-1526/1/artifact/out/patch-mvninstall-hadoop-ozone.txt |
| compile | https://builds.apache.org/job/hadoop-multibranch/job/PR-1526/1/artifact/out/patch-compile-hadoop-hdds.txt |
| compile | https://builds.apache.org/job/hadoop-multibranch/job/PR-1526/1/artifact/out/patch-compile-hadoop-ozone.txt |
| cc | https://builds.apache.org/job/hadoop-multibranch/job/PR-1526/1/artifact/out/patch-compile-hadoop-hdds.txt |
| cc | https://builds.apache.org/job/hadoop-multibranch/job/PR-1526/1/artifact/out/patch-compile-hadoop-ozone.txt |
| javac | https://builds.apache.org/job/hadoop-multibranch/job/PR-1526/1/artifact/out/patch-compile-hadoop-hdds.txt |
| javac | https://builds.apache.org/job/hadoop-multibranch/job/PR-1526/1/artifact/out/patch-compile-hadoop-ozone.txt |
| javadoc | https://builds.apache.org/job/hadoop-multibranch/job/PR-1526/1/artifact/out/patch-javadoc-hadoop-hdds.txt |
| javadoc | https://builds.apache.org/job/hadoop-multibranch/job/PR-1526/1/artifact/out/patch-javadoc-hadoop-ozone.txt |
| findbugs | https://builds.apache.org/job/hadoop-multibranch/job/PR-1526/1/artifact/out/patch-findbugs-hadoop-hdds.txt |
| findbugs | https://builds.apache.org/job/hadoop-multibranch/job/PR-1526/1/artifact/out/patch-findbugs-hadoop-ozone.txt |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-1526/1/artifact/out/patch-unit-hadoop-hdds.txt |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-1526/1/artifact/out/patch-unit-hadoop-ozone.txt |
|  Test Results | https://builds.apache.org/job/hadoop-multibranch/job/PR-1526/1/testReport/ |
| Max. process+thread count | 413 (vs. ulimit of 5500) |
| modules | C: hadoop-ozone/common hadoop-ozone/ozone-manager U: hadoop-ozone |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-1526/1/console |
| versions | git=2.7.4 maven=3.3.9 |
| Powered by | Apache Yetus 0.10.0 http://yetus.apache.org |


This message was automatically generated.

**Comment by hadoop-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 37 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | dupname | 0 | No case conflicting files found. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| +1 | test4tests | 0 | The patch appears to include 5 new or modified test files. |
||| _ trunk Compile Tests _ |
| 0 | mvndep | 14 | Maven dependency ordering for branch |
| -1 | mvninstall | 30 | hadoop-hdds in trunk failed. |
| -1 | mvninstall | 24 | hadoop-ozone in trunk failed. |
| -1 | compile | 20 | hadoop-hdds in trunk failed. |
| -1 | compile | 15 | hadoop-ozone in trunk failed. |
| +1 | checkstyle | 49 | trunk passed |
| +1 | mvnsite | 0 | trunk passed |
| +1 | shadedclient | 843 | branch has no errors when building and testing our client artifacts. |
| -1 | javadoc | 22 | hadoop-hdds in trunk failed. |
| -1 | javadoc | 20 | hadoop-ozone in trunk failed. |
| 0 | spotbugs | 943 | Used deprecated FindBugs config; considering switching to SpotBugs. |
| -1 | findbugs | 33 | hadoop-hdds in trunk failed. |
| -1 | findbugs | 20 | hadoop-ozone in trunk failed. |
||| _ Patch Compile Tests _ |
| 0 | mvndep | 19 | Maven dependency ordering for patch |
| -1 | mvninstall | 36 | hadoop-hdds in the patch failed. |
| -1 | mvninstall | 30 | hadoop-ozone in the patch failed. |
| -1 | compile | 24 | hadoop-hdds in the patch failed. |
| -1 | compile | 19 | hadoop-ozone in the patch failed. |
| -1 | cc | 24 | hadoop-hdds in the patch failed. |
| -1 | cc | 19 | hadoop-ozone in the patch failed. |
| -1 | javac | 24 | hadoop-hdds in the patch failed. |
| -1 | javac | 19 | hadoop-ozone in the patch failed. |
| +1 | checkstyle | 57 | the patch passed |
| +1 | mvnsite | 0 | the patch passed |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedclient | 722 | patch has no errors when building and testing our client artifacts. |
| -1 | javadoc | 21 | hadoop-hdds in the patch failed. |
| -1 | javadoc | 20 | hadoop-ozone in the patch failed. |
| -1 | findbugs | 32 | hadoop-hdds in the patch failed. |
| -1 | findbugs | 20 | hadoop-ozone in the patch failed. |
||| _ Other Tests _ |
| -1 | unit | 28 | hadoop-hdds in the patch failed. |
| -1 | unit | 24 | hadoop-ozone in the patch failed. |
| +1 | asflicense | 32 | The patch does not generate ASF License warnings. |
| | | 2356 | |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.1 Server=19.03.1 base: https://builds.apache.org/job/hadoop-multibranch/job/PR-1526/2/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/1526 |
| Optional Tests | dupname asflicense compile cc mvnsite javac unit javadoc mvninstall shadedclient findbugs checkstyle |
| uname | Linux becb47a90b3b 4.15.0-58-generic #64-Ubuntu SMP Tue Aug 6 11:12:41 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / bdaaa3b |
| Default Java | 1.8.0_222 |
| mvninstall | https://builds.apache.org/job/hadoop-multibranch/job/PR-1526/2/artifact/out/branch-mvninstall-hadoop-hdds.txt |
| mvninstall | https://builds.apache.org/job/hadoop-multibranch/job/PR-1526/2/artifact/out/branch-mvninstall-hadoop-ozone.txt |
| compile | https://builds.apache.org/job/hadoop-multibranch/job/PR-1526/2/artifact/out/branch-compile-hadoop-hdds.txt |
| compile | https://builds.apache.org/job/hadoop-multibranch/job/PR-1526/2/artifact/out/branch-compile-hadoop-ozone.txt |
| javadoc | https://builds.apache.org/job/hadoop-multibranch/job/PR-1526/2/artifact/out/branch-javadoc-hadoop-hdds.txt |
| javadoc | https://builds.apache.org/job/hadoop-multibranch/job/PR-1526/2/artifact/out/branch-javadoc-hadoop-ozone.txt |
| findbugs | https://builds.apache.org/job/hadoop-multibranch/job/PR-1526/2/artifact/out/branch-findbugs-hadoop-hdds.txt |
| findbugs | https://builds.apache.org/job/hadoop-multibranch/job/PR-1526/2/artifact/out/branch-findbugs-hadoop-ozone.txt |
| mvninstall | https://builds.apache.org/job/hadoop-multibranch/job/PR-1526/2/artifact/out/patch-mvninstall-hadoop-hdds.txt |
| mvninstall | https://builds.apache.org/job/hadoop-multibranch/job/PR-1526/2/artifact/out/patch-mvninstall-hadoop-ozone.txt |
| compile | https://builds.apache.org/job/hadoop-multibranch/job/PR-1526/2/artifact/out/patch-compile-hadoop-hdds.txt |
| compile | https://builds.apache.org/job/hadoop-multibranch/job/PR-1526/2/artifact/out/patch-compile-hadoop-ozone.txt |
| cc | https://builds.apache.org/job/hadoop-multibranch/job/PR-1526/2/artifact/out/patch-compile-hadoop-hdds.txt |
| cc | https://builds.apache.org/job/hadoop-multibranch/job/PR-1526/2/artifact/out/patch-compile-hadoop-ozone.txt |
| javac | https://builds.apache.org/job/hadoop-multibranch/job/PR-1526/2/artifact/out/patch-compile-hadoop-hdds.txt |
| javac | https://builds.apache.org/job/hadoop-multibranch/job/PR-1526/2/artifact/out/patch-compile-hadoop-ozone.txt |
| javadoc | https://builds.apache.org/job/hadoop-multibranch/job/PR-1526/2/artifact/out/patch-javadoc-hadoop-hdds.txt |
| javadoc | https://builds.apache.org/job/hadoop-multibranch/job/PR-1526/2/artifact/out/patch-javadoc-hadoop-ozone.txt |
| findbugs | https://builds.apache.org/job/hadoop-multibranch/job/PR-1526/2/artifact/out/patch-findbugs-hadoop-hdds.txt |
| findbugs | https://builds.apache.org/job/hadoop-multibranch/job/PR-1526/2/artifact/out/patch-findbugs-hadoop-ozone.txt |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-1526/2/artifact/out/patch-unit-hadoop-hdds.txt |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-1526/2/artifact/out/patch-unit-hadoop-ozone.txt |
|  Test Results | https://builds.apache.org/job/hadoop-multibranch/job/PR-1526/2/testReport/ |
| Max. process+thread count | 446 (vs. ulimit of 5500) |
| modules | C: hadoop-ozone/common hadoop-ozone/ozone-manager U: hadoop-ozone |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-1526/2/console |
| versions | git=2.7.4 maven=3.3.9 |
| Powered by | Apache Yetus 0.10.0 http://yetus.apache.org |


This message was automatically generated.

**Comment by anuengineer** [comment]
> Was this committed without a pre-commit run from Anzix?

Looks like it was not run. I did look at the yetus output but the output is too noisy to make any sense.

**Comment by arp7** [comment]
None that I know of. However we should not commit a patch without CI on principle.

**Comment by anuengineer** [comment]
> None that I know of. However we should not commit a patch without CI on principle.

This has been broken for quite a while; So I rely on Jenkins/Yetus and hand building before apply the commits.

**Pull request merged**